src/intersections/lines.jl

```diff
@@ -43,14 +28,38 @@ end
 function intersectpoint(l1::Line, l2::Line)
   a, b = l1(0), l1(1)
   c, d = l2(0), l2(1)
+  λ₁, _ = intersectparameters(a, b, c, d)
+  a + λ₁ * (b - a)
+ end
 
-  v1  = a - b
-  v2  = c - d
-  v12 = a - c
+# compute the parameters of the lines defined by the points a -> b and c -> d 
+# so that the resulting points have minimal distance
+# the ranks help to identify the different types of intersection
+function intersectparameters(a::Point{N,T}, b::Point{N,T}, 
+                             c::Point{N,T}, d::Point{N,T}) where {N,T}
+  # solves the equation (approximately):
+  # a + λ₁ ⋅ v⃗₁ = c + λ₂ ⋅ v⃗₂,
+  # with v⃗₁ = b - a, v⃗₂ = d - c
+  A = [(b - a) (c - d)]
+  y = c - a
+  QR = qr([A y])
 
-  # the intersection point lies in between a and b at a fraction s
-  # (https://en.wikipedia.org/wiki/Line-line_intersection#Formulas)
-  s = (v12[1] * v2[2] - v12[2] * v2[1]) / (v1[1] * v2[2] - v1[2] * v2[1])
+  # calculate the rank of the augmented matrix
+  rₐ = sum(sum(abs.(QR.R), dims = 2) .> atol(T))
```

Comment:
```suggestion
  rₐ = sum(sum(abs, QR.R, dims = 2)) .> atol(T))
```

Perhaps you can express this in terms of the actual norm you are interested? The `norm` function isn't useful here?

Reply:
To evaluate the rank of this rectangular matrix, I count how many rows are non-zero, so how many row norms are above the tolerance.
I don't see how norm can help me since there is no `dims` option. 
I will adopt your suggestion to avoid the syntax error:
`rₐ = sum(sum(abs, QR.R; dims = 2) .> atol(T))`

Reply:
I just figured out that my algorithm doesn't yield the right rank for all cases:

Here are the two examples I struggle with: `R` matrices from QR decompositions
``` julia
[ -3.31662   3.31662      -0.301511
   0.0      -4.44089e-16   0.904534
   0.0       0.0           0.301511 ]

[ -1.0  1.0  0.0
   0.0  0.0  1.0 ]
```
A solution to evaluate the rank correctly would be:
``` code
rₐ = N == 2 ? sum(sum(abs, QRₐ.R; dims = 2) .> atol(T)) :
              sum(abs.(diag(QRₐ.R)) .> atol(T))
```
Maybe you have a more elegant solution?

Reply:
After you refactor the code to use norms we can see if there is a mathematical identity to be explored.